Reorganize getting-started + added a new tab "Core Concepts" (#24)
php-etl/documentation: content/getting-started/execution/_index.en.md

---
title: "Execution"
date: 2023-02-14T15:33:44+01:00
draft: true
weight: 5
---

Once your PHP code is generated, it can be executed right away.

> n the following paragraphs, you will find the method for executing the satellites depending on 

Reviewer: ```suggestion
In the following paragraphs, you will find the method for executing the satellites depending on 
```

Author: Modifié

---
title: "Execution"
date: 2023-02-14T15:33:44+01:00
draft: true
weight: 5
---

> Once your PHP code is generated, you only need to execute it.

Reviewer 2: ```suggestion
Once your PHP code is generated, it can be executed right away.
```

‏

> However, depending on the adapter you have chosen in your configuration, the way to do it is different.

Reviewer 2: **💭 thought:** On ne devrait pas renvoyer sur la doc de ce qu'est un adapter ?‏

Author: J'ai rajouté le lien

> However, depending on the adapter you have chosen in your configuration, the way to do it is different.

Reviewer 2: ```suggestion
In the following paragraphs, you will find the method for executing the satellites depending on the adapter you have chosen.
```

## If you built inside the filesystem

```shell
# php <path-to-the-generated-satellite>
php build/main.php
```

This command will run your satellite service located in a directory. The execution of your service may take several minutes,

> so please wait until the execution is completed.

Reviewer 2: **💭 thought:** La commande satellite a des commandes pour lancer les pipelines.

Author: Effectivement, j'ai remplacé par `php bin/satellite run:pipeline build/`

> so please wait until the execution is completed.

Reviewer 2: **💭 thought:** est-ce qu'on n'en profiterait pas pour mettre des captures d'écran ?‏

## If you built as a Docker image

```shell
# docker run --rm -ti <generated-satellite-docker-image-name>
docker run --rm -ti foo/satellite-name:latest
```

> This command will run your satellite service located in a Docker image. The execution of your service may take several minutes,
> so please wait until the execution is completed.

Reviewer 2: ```suggestion
This command will run your satellite service you have previously built as a Docker image you named `foo/satellite-name:latest`.
```
